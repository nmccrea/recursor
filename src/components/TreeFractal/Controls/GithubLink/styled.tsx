import styled from "styled-components"
import githubLogoUrl from "images/icon-logo-github-mark.png"

export const GithubImage = styled.div`
  background-image: url(${githubLogoUrl});
  background-size: 1.6rem 1.6rem;
  width: 1.6rem;
  height: 1.6rem;
  opacity: 0.6;
  margin-right: 0.5rem;

  &:hover {
    opacity: 0.8;
  }
`
