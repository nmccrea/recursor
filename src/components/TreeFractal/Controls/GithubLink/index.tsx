import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import { GithubImage } from "./styled"

const GithubLink = () => {
  const { site } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            githubUrl
          }
        }
      }
    `
  )

  return (
    <a href={site.siteMetadata.githubUrl} target="blank">
      <GithubImage />
    </a>
  )
}

export default GithubLink
