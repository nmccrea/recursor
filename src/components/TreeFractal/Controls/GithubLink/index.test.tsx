import { render } from "@testing-library/react"
import "jest-styled-components"
import React from "react"
import GithubLink from "."

describe("`<GithubLink />`", () => {
  it("renders correctly", () => {
    const { container } = render(<GithubLink />)

    expect(container).toMatchSnapshot()
  })
})
