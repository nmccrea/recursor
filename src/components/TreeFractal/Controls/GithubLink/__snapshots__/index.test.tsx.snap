// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<GithubLink />\` renders correctly 1`] = `
.c0 {
  background-image: url(-- MOCK FILE EXPORT VALUE --);
  background-size: 1.6rem 1.6rem;
  width: 1.6rem;
  height: 1.6rem;
  opacity: 0.6;
  margin-right: 0.5rem;
}

.c0:hover {
  opacity: 0.8;
}

<div>
  <a
    href="https://github.com/mockiemcgee/test"
    target="blank"
  >
    <div
      class="c0"
    />
  </a>
</div>
`;
